TrackerExtenstion.cs

```diff
@@ -38,14 +39,48 @@ public void Startup(StartupParams sp)
         /// </param>
         public void Ready(ReadyParams rp)
         {
+            // @todo Subscribe to an event after the Dynamo home screen is shown
+
+            // Display a MessageBox to the user
+            // @todo Use a view framework to improve the UI/UX
             string message = "By pressing OK you agreeing to Binoculars 🔍 data collection.";
             string title = "Terms of Use Agreement";
-
             MessageBox.Show(message, title, MessageBoxButton.OK, MessageBoxImage.Exclamation);
+
+            // @todo if user does not consent, don't store..
+            if (true) {
+
+                // Dynamo StartupParams
+                ExportData.dynamoVersion = rp.StartupParams.DynamoVersion.ToString();
+
+                // @todo Determine whether the script is run from DynamoPlayer..
+
+                // Store the local environment data
+                ExportData.user = Environment.MachineName;
+                ExportData.computerName = Environment.UserName;
+                
+                // @todo Define the revitVersion, leave blank or null if opened from any other environment (Sandbox, Civil3D etc)
+                ExportData.revitVersion = "2019.0.2";
+
+                // Request the IP and geolocation from ipinfo.io
+                WebClient client = new WebClient();
+                client.Headers.Set("Accept", "application/json");
+                var json = client.DownloadString("https://ipinfo.io");
+                // @todo We need to check the request was actually sent successfully and gracefully deal with cases where the API is unavailable
+
+                // Parse the JSON
+                JObject ipinfo = JObject.Parse(json);
+
+                // Store geolocation data
+                ExportData.ip = (string)ipinfo["ip"];
+                ExportData.latlng = (string)ipinfo["loc"];
+                ExportData.city = (string)ipinfo["city"];
+                ExportData.country = (string)ipinfo["country"];
+            }
             
             // we can register our own events that will be triggered when specific things happen in Dynamo
             // a reference to the ReadyParams is needed to do this, so we pass it on
-            TrackerEvents.RegisterEventHandlers(rp);            
+            TrackerEvents.RegisterEventHandlers(rp);
```

Comment:
if this stays here, the `if` that bails early would mean this never gets triggered - which i assume is the desired behaviour given someone hasn't opted in.

Reply:
Closing in favour of issue https://github.com/teamtreedyn/Binoculars/issues/20